finality-aleph/src/metrics/finality_rate.rs

```diff
@@ -0,0 +1,162 @@
+use std::collections::{hash_map::Entry, HashMap};
+
+use primitives::{BlockHash, BlockNumber};
+use substrate_prometheus_endpoint::{register, Counter, PrometheusError, Registry, U64};
+
+#[derive(Clone)]
+pub enum FinalityRateMetrics {
+    Prometheus {
+        own_finalized: Counter<U64>,
+        own_hopeless: Counter<U64>,
+        imported_cache: HashMap<BlockNumber, Vec<BlockHash>>,
```

Comment:
nit: this vec should be set actually

Reply:
Why tho? We don't ever search for any element in this collection - only increase it / iterate through its elements once.

Reply:
its a small nit. The iteration could e replaced instead with finding the element in the collection plus the hashes here should be unique. Feel free to resolve